actionpack/lib/action_controller/renderer.rb

```diff
@@ -110,16 +121,24 @@ def render(*args)
     alias_method :render_to_string, :render # :nodoc:
 
     private
-      def normalize_keys(defaults, env)
+      def self.normalize_env(env)
```

Comment:
This method isn't private since it is now a class method. We need to either use `class << self` or call `private_class_method` But it seems you want it to be public since we are using delegation, so let's just move it out of the `private` section.